common/freezer.ts

```diff
@@ -0,0 +1,176 @@
+import { spawn } from 'child_process';
+import * as path from 'path';
+import * as fs from 'fs';
+
+const PROJECT_BASE = path.resolve('./');
+const ORACLE_BRANCH = 'develop';
+const GET_PACKAGE_CMD = `git show ${ORACLE_BRANCH}:package.json`;
+const GET_DIFF_CMD = `git diff origin/${ORACLE_BRANCH}`;
+
+const newFileRegEx = /^\+\+\+ b\//;
+const frozenFolderRegEx = /\/\*$/;
+
+const start = async () => {
+  try {
+    const packageStr = await runShCommand(GET_PACKAGE_CMD);
+    const diff = await runShCommand(GET_DIFF_CMD);
+    const { frozen } = JSON.parse(packageStr);
+
+    if (frozen === undefined) {
+      console.log(
+        `Freezer: No config found in package.json on branch ${
+          ORACLE_BRANCH
+        }. Exiting.`
+      );
+      return;
+    }
+
+    const newFiles = getNewFiles(diff);
+    const frozenFiles = getFrozenFiles(frozen);
+    const frozenFolders = getFrozenFolders(frozen);
+
+    ensureNewFilesAreNotFrozen(newFiles, frozenFiles, frozenFolders);
+  } catch (err) {
+    console.log(err.message);
+    exit();
```

Comment:
This should be an `exit(1)` to ensure that the build fails if something's wrong with freezer.

Reply:
The `exit` here is a constant that calls `process.exit(1)`, ensuring a failing build on error.